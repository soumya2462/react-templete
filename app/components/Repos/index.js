/**
 *
 * Repos
 *
 */

import Card from '@components/Card';
import For from '@components/For';
import If from '@components/If';
import RepoCard from '@components/RepoCard';
import { Skeleton } from 'antd';
import { get, isEmpty } from 'lodash';
import PropTypes from 'prop-types';
import React from 'react';
import { FormattedMessage as T } from 'react-intl';
import styled from 'styled-components';

const Container = styled.div`
  && {
    display: flex;
    flex-direction: column;
    max-width: ${(props) => props.maxwidth}px;
    width: 100%;
    margin: 0 auto;
    padding: ${(props) => props.padding}px;
  }
`;

const Repos = ({ data, loading, repo }) => {
  const items = get(data, 'items', []);
  const totalCount = get(data, 'totalCount', 0);

  return (
    <If condition={!isEmpty(items) || loading}>
      <Card data-testid="repos">
        <Skeleton loading={loading} active>
          <If condition={!isEmpty(repo)}>
            <div>
              <T id="search_query" values={{ repoName: repo }} />
            </div>
          </If>
          <If condition={totalCount !== 0}>
            <div>
              <T id="matching_repos" values={{ totalCount }} />
            </div>
          </If>
          <>
            <For
              of={items}
              ParentComponent={Container}
              renderItem={(item, index) => <RepoCard key={index} name={item.name} />}
            />
          </>
        </Skeleton>
      </Card>
    </If>
  );
};

const ReposShape = PropTypes.shape({
  totalCount: PropTypes.number,
  incompleteResults: PropTypes.bool,
  items: PropTypes.array
});

Repos.propTypes = {
  repo: PropTypes.string,
  data: ReposShape,
  loading: PropTypes.bool
};

export default Repos;
