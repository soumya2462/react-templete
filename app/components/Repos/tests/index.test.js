/**
 *
 * Tests for Repos
 *
 */

import React from 'react';
// import { fireEvent } from '@testing-library/dom'
import { renderWithIntl } from '@utils/testUtils';
import Repos from '../index';

const data = { items: [{ name: 'kuntham' }] };

describe('<Repos />', () => {
  it('should render and match the snapshot', () => {
    const { baseElement } = renderWithIntl(<Repos data={data} />);
    expect(baseElement).toMatchSnapshot();
  });

  it('should contain 1 Repos component', () => {
    const { getAllByTestId } = renderWithIntl(<Repos data={data} />);
    expect(getAllByTestId('repos').length).toBe(1);
  });
});
