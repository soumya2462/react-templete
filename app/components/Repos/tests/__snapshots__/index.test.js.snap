// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Repos /> should render and match the snapshot 1`] = `
<body>
  <div>
    <div
      class="ant-card ant-card-bordered Card__CustomCard-sc-7y2i64-0 hWoWqx"
      data-testid="repos"
    >
      <div
        class="ant-card-body"
      >
        <div
          class="Repos__Container-sc-1y64nrh-0 iMgkMF"
          data-testid="for"
          orientation="row"
        >
          <div
            class="ant-card ant-card-bordered Card__CustomCard-sc-7y2i64-0 hWoWqx"
            data-testid="repo-card"
          >
            <div
              class="ant-card-body"
            >
              <p
                class="T__StyledText-gjlic1-0 kYcyRq"
                data-testid="name"
              >
                Repository Name: kuntham
              </p>
              <p
                class="T__StyledText-gjlic1-0 kYcyRq"
                data-testid="fullName-unavailable"
              >
                Repository full name unavaiable
              </p>
              <p
                class="T__StyledText-gjlic1-0 kYcyRq"
                data-testid="stargazers-unavaiable"
              >
                Repository stars are unavaiable
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
